contracts/p1/StRSR.sol

```diff
@@ -335,6 +335,54 @@ abstract contract StRSRP1 is Initializable, ComponentP1, IStRSR, EIP712Upgradeab
         IERC20Upgradeable(address(rsr)).safeTransfer(account, rsrAmount);
     }
 
+    function cancelUnstake(uint256 endId) external notPausedOrFrozen {
+        address account = _msgSender();
+
+        // == Refresh ==
+        assetRegistry.refresh();
+
+        // == Checks + Effects ==
+        require(basketHandler.fullyCollateralized(), "RToken uncollateralized");
+        require(basketHandler.status() == CollateralStatus.SOUND, "basket defaulted");
+
+        uint256 firstId = firstRemainingDraft[draftEra][account];
+        CumulativeDraft[] storage queue = draftQueues[draftEra][account];
+        if (endId == 0 || firstId >= endId) return;
```

Comment:
So, just because `withdraw()` does this doesn't mean `cancelUnstake()` has to. I do prefer the revert behavior myself. 

I would also check to see what it breaks in tests to use revert in `withdraw()` as well. If it's not much, maybe it's worth doing.